python/cudf/cudf/core/frame.py

```diff
@@ -4826,13 +4828,16 @@ def nans_to_nulls(self):
         1  <NA>  3.14
         2  <NA>  <NA>
         """
-        return self._from_data(
-            {
-                name: col.copy().nans_to_nulls()
-                for name, col in self._data.items()
-            },
-            self._index,
-        )
+        result_data = {}
+        for name, col in self._data.items():
+            if (
+                isinstance(col, cudf.core.column.NumericalColumn)
+                and col.nan_count
```

Comment:
This makes me think should we have a `nan_count` property defined in `ColumnBase` to return `0`? @shwina @vyasr

Reply:
I think:

- The check for  `nan_count`  should be in `NumericalColumn.nans_to_nulls()`
- Rather than the `isinstance` check, we could do a `try...except`:

```python
try:
    result_data[name] = col.nans_to_nulls()
except AttributeError:
    result_data[name] = col.copy()
else:
    result_data[name] = col.copy()
```